homeassistant/components/fireservicerota/switch.py

```diff
@@ -0,0 +1,154 @@
+        self._entry_id = entry.entry_id
+
+        self._state = None
+        self._state_attributes = {}
+        self._state_icon = None
+
+    @property
+    def name(self) -> str:
+        """Return the name of the switch."""
+        return "Incident Response"
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        if self._state_icon == "acknowledged":
+            return "mdi:run-fast"
+        if self._state_icon == "rejected":
+            return "mdi:account-off-outline"
+
+        return "mdi:forum"
+
+    @property
+    def is_on(self) -> bool:
+        """Get the assumed state of the switch."""
+        return self._state
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this switch."""
+        return self._unique_id
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def available(self):
+        """Return if switch is available."""
+        return self._client.on_duty
+
+    @property
+    def device_state_attributes(self) -> object:
+        """Return available attributes for switch."""
+        attr = {}
+        if not self._state_attributes:
+            return attr
+
+        data = self._state_attributes
+        attr = {
+            key: data[key]
+            for key in (
+                "user_name",
+                "assigned_skill_ids",
+                "responded_at",
+                "start_time",
+                "status",
+                "reported_status",
+                "arrived_at_station",
+                "available_at_incident_creation",
+                "active_duty_function_ids",
+            )
+            if key in data
+        }
+
+        _LOGGER.debug("Set attributes of entity 'Response Switch' to '%s'", attr)
+        return attr
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Send Acknowlegde response status."""
+        await self.async_set_response(True)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Send Reject response status."""
+        await self.async_set_response(False)
+
+    async def async_set_response(self, value) -> None:
+        """Send response status."""
+        if not self._client.on_duty:
+            _LOGGER.debug(
+                "Cannot send incident response when not on duty",
+            )
+            return
+
+        await self._client.async_set_response(value)
+        self.client_update()
+
+    async def async_added_to_hass(self) -> None:
+        """Register update callback."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{FIRESERVICEROTA_DOMAIN}_{self._entry_id}_update",
+                self.client_update,
+            )
+        )
+        self.async_on_remove(self._coordinator.async_add_listener(self.on_duty_update))
+
+    @callback
+    def on_duty_update(self):
```

Comment:
We can connect `self.async_write_ha_state` directly to the coordinator.

Reply:
@MartinHjelmare 

Do you mean calling this?
`self._coordinator.async_write_ha_state`
This doesn't update the entity state it seems. Please elaborate, the coordinator is new for me.

Reply:
BTW I have created PR #43741 to follow up on almost al of your comments, a few left.

Reply:
Like so:
```py
self.async_on_remove(self._coordinator.async_add_listener(self.async_write_ha_state))
```

Reply:
Ah, yeah, less code to do the same is always nice.
Implemented.